src/main/java/com/textmessenger/service/UserServiceImpl.java

```diff
@@ -234,4 +244,70 @@ public String changePasswordForgot(CredentialsPassword credentialsPassword) {
     }
     return "Sorry. This token in invalid.";
   }
+
+  @Override
+  public void updateUserWithStringsAndFile(String firstName,
+                                           String lastName,
+                                           String address,
+                                           String date,
+                                           MultipartFile file) throws IOException {
+    UserPrincipal userPrincipal = (UserPrincipal) SecurityContextHolder
+            .getContext()
+            .getAuthentication()
+            .getPrincipal();
+    User one = userRepository.getOne(userPrincipal.getId());
+    if (firstName != "undefined") {
+      one.setFirstName(firstName);
+    }
+    if (lastName != "undefined") {
+      one.setLastName(lastName);
+    }
+    if (address != "undefined") {
+      one.setAddress(address);
+    }
+    if (file != null) {
+      String typeFile = file.getContentType();
+      String type = "." + typeFile.substring(6);
+      String key = "userAvatar/" + UUID.randomUUID() + type;
+      InputStream fileFromFront = file.getInputStream();
+      AmazonS3 amazonS3 = s3.getConnection();
+      amazonS3.putObject(
+              BUCKET,
+              key,
+              fileFromFront,
+              new ObjectMetadata());
+      String userHeader = amazonS3.getUrl(BUCKET, key).toString();
+      if (one.getProfilePhoto() != null) {
+        //TODO logic of remove and set new photo
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/final_project_text_messenger/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=Text_Messenger_Project&pullRequest=194&issues=AWXd3GSh_eEZPtUmH9zI&open=AWXd3GSh_eEZPtUmH9zI)

[ISSUE_KEY]: AWXd3GSh_eEZPtUmH9zI